src/components/signin/IOSWebAppSignIn.js

```diff
@@ -0,0 +1,146 @@
+    setRecovering(true)
+
+    const errorText = 'You are using wrong sign in code'
+    let userNameAndPWD = Buffer.from(code, 'base64').toString('ascii')
+    let userNameAndPWDArray = userNameAndPWD.split('+')
+
+    log.debug('IOSSignInByMagicCode', { code, userNameAndPWDArray })
+
+    if (userNameAndPWDArray.length === 2) {
+      const userName = userNameAndPWDArray[0]
+      const userPwd = userNameAndPWDArray[1]
+      const UserStorage = await import('../../lib/gundb/UserStorageClass').then(_ => _.UserStorage)
+
+      const mnemonic = await UserStorage.getMnemonic(userName, userPwd)
+
+      if (mnemonic && bip39.validateMnemonic(mnemonic)) {
+        const mnemonicsHelpers = import('../../lib/wallet/SoftwareWalletProvider')
+        const { saveMnemonics } = await mnemonicsHelpers
+
+        await saveMnemonics(mnemonic)
+        await AsyncStorage.setItem(IS_LOGGED_IN, true)
+
+        window.location = '/'
+      } else {
+        showErrorDialog(errorText)
+      }
+    } else {
+      showErrorDialog(errorText)
+    }
+
+    setRecovering(false)
+  }
+
+  const handleEnter = async (event: { nativeEvent: { key: string } }) => {
+    if (event.nativeEvent.key === 'Enter' && isValid) {
+      await recover()
+    }
+  }
+
+  return (
+    <Fragment>
+      <NavBar title={TITLE} />
+      <Section grow={5} style={styles.wrapper}>
+        <Section.Stack grow style={styles.instructions} justifyContent="center">
+          <Text fontWeight="medium" fontSize={22} fontFamily="Roboto" lineHeight={28} style={styles.headerText}>
+            {'Easy sign in with iPhone!'}
+          </Text>
+          <Text color="gray80Percent" fontSize={14}>
+            {'Copy-paste the iPhone code which\nwas sent to your phone on SMS:'}
+          </Text>
+        </Section.Stack>
+        <Section.Stack justifyContent="space-between">
+          <Section.Row justifyContent="center">
+            <InputText
+              value={code}
+              onChangeText={handleChange}
+              onKeyPress={handleEnter}
+              onCleanUpField={handleChange}
+              autoFocus
+            />
+          </Section.Row>
+        </Section.Stack>
+        <Image source={IOSWebAppSignInSVG} resizeMode={'contain'} style={styles.image} />
+        <Section.Stack grow style={styles.bottomContainer} justifyContent="flex-end">
+          <CustomButton style={styles.buttonLayout} onPress={recover} disabled={!isValid || isRecovering}>
+            {'SIGN IN'}
+          </CustomButton>
+        </Section.Stack>
+      </Section>
+    </Fragment>
+  )
+}
+
+Mnemonics.navigationOptions = {
+  title: TITLE,
+}
+
+const mnemonicsStyles = ({ theme }) => ({
+  headerText: {
+    marginBottom: getDesignRelativeHeight(5),
+  },
+  wrapper: {
+    borderRadius: 0,
+    paddingVertical: 0,
+  },
+  instructions: {
+    marginTop: getDesignRelativeHeight(10),
+  },
+  buttonLayout: {
+    marginVertical: 20,
+  },
+  bottomContainer: {
+    maxHeight: 80,
```

Comment:
relative height? why 80? we need the screen to fit nicely on iphone 5

Reply:
done. fixed to look nice on iphone 5